proxy/start_container_interceptor.go

```diff
@@ -35,7 +35,7 @@ func (i *startContainerInterceptor) InterceptRequest(r *http.Request) error {
 		}
 	}
 
-	if err := marshalRequestBody(r, hostConfig); err != nil {
+	if err := marshalRequestBody(r, *hostConfig); err != nil {
```

Comment:
Uhm, it is not completely equivalent (note that the reason for this PR was precisely that unmarshalling JSON leads to a different result for pointers and structs).

Probably in this case it doesn't make a difference (we guarantee that the pointer is not null), but I did it to make it explicit.